src/base/cub_ddl_log.c

```diff
@@ -0,0 +1,1054 @@
+  char *tpath = NULL;
+  int retval = 0;
+
+  if (ddl_audit_handle == NULL || file_full_path == NULL || dest_path == NULL || buf_size < 0)
+    {
+      return -1;
+    }
+
+  env_root = envvar_root ();
+  cub_get_current_time (time, TIME_STRING_SIZE);
+
+  name_tmp = strrchr (file_full_path, PATH_SEPARATOR);
+
+#if defined(WINDOWS)
+  {
+    const char *nn_tmp = strrchr (file_full_path, PATH_SEPARATOR);
+    if (name_tmp < nn_tmp)
+      {
+	name_tmp = nn_tmp;
+      }
+  }
+#endif /* WINDOWS */
+  if (name_tmp == NULL)
+    {
+      name_tmp = file_full_path;
+    }
+  else
+    {
+      name_tmp++;
+    }
+
+  retval = snprintf (ddl_audit_handle->schema_file, buf_size, "%s_%s_%d", name_tmp, time, getpid ());
+  if (retval < 0)
+    {
+      assert (false);
+      ddl_audit_handle->schema_file[0] = '\0';
+      return retval;
+    }
+
+  retval =
+    snprintf (dest_path, buf_size, "%s/%s/%s_%s_%d", env_root, DDL_LOG_LOADDB_FILE_PATH, name_tmp, time, getpid ());
+
+  if (retval < 0)
+    {
+      assert (false);
+      dest_path[0] = '\0';
+      return retval;
+    }
+#if defined(WINDOWS)
+  unix_style_path (dest_path);
+#endif
+
+  tpath = strdup (dest_path);
+  if (tpath == NULL)
+    {
+      return -1;
+    }
+  cub_create_dir_log (tpath);
+  FREE (tpath);
+
+  return retval;
+}
+
+static int
+cub_make_ddl_log_filename (char *filename_buf, size_t buf_size, const char *app_name)
+{
+  const char *env_root = NULL;
+  int retval = 0;
+
+  assert (filename_buf != NULL);
+
+  env_root = envvar_root ();
+
+  if (strcmp (app_name, "cas") == 0)
+    {
+      retval =
+	snprintf (filename_buf, buf_size, "%s/%s/%s_%d.ddl.log", env_root, DDL_LOG_PATH, ddl_audit_handle->br_name,
+		  (ddl_audit_handle->br_index + 1));
+    }
+  else if (strcmp (app_name, "csql") == 0 || strcmp (app_name, "loaddb") == 0)
+    {
+      retval =
+	snprintf (filename_buf, buf_size, "%s/%s/%s_%s_ddl.log", env_root, DDL_LOG_PATH, app_name,
+		  ddl_audit_handle->db_name);
+    }
+  else
+    {
+      filename_buf[0] = '\0';
+      return -1;
+    }
+
+  if (retval < 0)
+    {
+      assert (false);
+      filename_buf[0] = '\0';
+    }
+#if defined(WINDOWS)
+  unix_style_path (filename_buf);
+#endif
+  return retval;
+}
+
+static FILE *
+cub_ddl_log_open (char *app_name)
+{
+  FILE *fp = NULL;
+  char *tpath = NULL;
+  int len;
+
+  if (app_name != NULL)
+    {
+      len = cub_make_ddl_log_filename (ddl_audit_handle->log_filepath, PATH_MAX, app_name);
+
+      if (ddl_audit_handle->log_filepath[0] == '\0' || len < 0)
+	{
+	  goto file_error;
+	}
+
+      tpath = strdup (ddl_audit_handle->log_filepath);
+      if (tpath == NULL)
+	{
+	  goto file_error;
+	}
+      cub_create_dir_log (tpath);
+      FREE_MEM (tpath);
+    }
+  else
+    {
+      goto file_error;
+    }
+
+  /* note: in "a+" mode, output is always appended */
+  fp = cub_fopen_and_lock (ddl_audit_handle->log_filepath, "a+");
+
+  if (fp != NULL)
+    {
+      fseek (fp, 0, SEEK_END);
+    }
```

Comment:
Since you have already opened the file in "a +" mode, fp will be at the end of the file.
So, I don't think fseek() is necessary.

Reply:
Thank you, i will change code as your recommend.